### PR TITLE
Build changes

### PR DESCRIPTION
**Pull request by jbrooksuk**

### CODE REVIEW

File: build/md2json.js [hunk 1]
```diff
@@ -3,129 +3,143 @@ fs = require('fs')
...
-    console.log(JSON.stringify(root));
-  });
+
+        console.log(JSON.stringify(root));
+    });
 }
 
 if (process.argv.length < 3) {
     console.log("No .md file passed!");
+
```
Comment (davemachado) [inline build/md2json.js]: do we need this extra line to separate a simple "print and return"?

File: build/md2json.js [hunk 2]
```diff
@@ -3,129 +3,143 @@ fs = require('fs')
...
-                } else {
-                    row.push(line_this[j]);
+            while (line.indexOf("|") != -1 && cur_line != lines.length) {
+                var line_this = line.split("|")
+                var row = []
+
+                for (var j in line_this) {
+                    line_this[j] = md_trim(line_this[j], j)
+                    if ((j == 0 || j == line_this.length - 1) && line_this[j] === "") {
+                        //
```
Comment (davemachado) [inline build/md2json.js]: `// invalid line segment`

File: build/md2json.js [hunk 3]
```diff
@@ -3,129 +3,143 @@ fs = require('fs')
...
+            line = read_line()
 
-                } else {
-                    row.push(line_this[j]);
+            while (line.indexOf("|") != -1 && cur_line != lines.length) {
+                var line_this = line.split("|")
+                var row = []
+
+                for (var j in line_this) {
+                    line_this[j] = md_trim(line_this[j], j)
```
Comment (davemachado) [inline build/md2json.js]: md_trim(line_this[ j ], **parseInt**( j ))

File: build/md2json.js [hunk 4]
```diff
@@ -3,129 +3,143 @@ fs = require('fs')
...
-    function read_line() {
-        return lines[cur_line++];
-    }
-    var root = {};
-    while (true) {
+        if (err) {
+            return console.log(err);
+        }
+
+        var lines = text.split("\n");
```
Comment (jbrooksuk) [inline build/md2json.js]: Yes! This was going to be my next step but I had a flight to catch 🙂

File: build/md2json.js [hunk 5]
```diff
@@ -3,129 +3,143 @@ fs = require('fs')
...
+        var table_name = "";
+        var col_num = 0;
         var cols = [];
         var rows = [];
-        while (line.indexOf(anchor) == -1 && cur_line != lines.length) {
-            line = read_line();
-        }
-        if (cur_line == lines.length) {
-            break;
+        var root = {};
```
Comment (jbrooksuk) [inline build/md2json.js]: Yeah, another thing on my after flight list ;)

**Comment by mikestreety** [comment]
Is this waiting for @jbrooksuk? No rush, just trying to work out the next step :)

**Comment by mikestreety** [comment]
@jbrooksuk @davemachado Nudge 😉

**Comment by jbrooksuk** [comment]
Yeah, it's all on me. Will get there, promise :)

**Comment by davemachado** [comment]
Looks like @jbrooksuk gets a free pass on this PR since #569 removes all these files 😄

**Pull request closed without merging**